tests: drivers: build_all: Add power_domain drivers test

soburi opened the pull request:
Add build_all test for power_domain drivers.

soburi committed:
drivers: power_domain: gpio_monitor: Fix compilation error

The fix 25173f71cda630d4fb0c860d4e45e6f93c0995dd created
`struct pm_device_base` and moved `usage` to it.
This has been overlooked so that I fixed it.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

soburi committed:
tests: drivers: build_all: Add power_domain drivers test

Add build_all test for power_domain drivers.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

JordanYates reviewed (changes requested):
Please also add an instance of `power-domain-gpio`, and add a test variant with PM enabled like this:
```
 drivers.power_domain.build.pm: 
   extra_configs: 
     - CONFIG_PM_DEVICE=y 
     - CONFIG_PM_DEVICE_RUNTIME=y 
```

JordanYates commented on tests/drivers/build_all/power_domain/prj.conf:
```diff
@@ -0,0 +1,6 @@
+CONFIG_TEST_USERSPACE=y
+CONFIG_GPIO=y
+CONFIG_POWER_DOMAIN=y
+CONFIG_PM_DEVICE=y
```
`PM_DEVICE` and `PM_DEVICE_RUNTIME` should not be added here, as they are enabled in the `drivers.power_domain.build.pm` testcase.

soburi replied:
Thank you for your check.
I've fixed it.
"power_domain,gpio_monitor" completely depends on PM_DEVICE, so I also fixed Kconfig.

soburi committed:
drivers: power_domain: gpio_monitor: Add PM_DEVICE dependency

Since "power_domain,gpio_monitor" depends on PM_DEVICE,
specify it in Kconfig.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>
